controllers/discordactions.js

```diff
@@ -200,35 +201,64 @@ const updateDiscordNicknames = async (req, res) => {
       });
     }
 
-    const discordServerUsers = await dataAccess.retrieveDiscordUsers();
-    const nonSuperUsers = discordServerUsers.filter((user) => !user.roles.super_user);
+    const membersInDiscord = await getDiscordMembers();
+    const usersToBeEffected = [];
+    await Promise.all(
+      membersInDiscord.map(async (discordUser) => {
+        try {
+          const foundUserWithDiscordId = await fetchUser({ discordId: discordUser.user.id });
+
+          if (foundUserWithDiscordId.userExists) {
+            const isDeveloper = discordUser.roles.includes(discordDeveloperRoleId);
+            const isMaven = discordUser.roles.includes(discordMavenRoleId);
+            const isBot = discordUser.user.bot;
+            const isUsernameMatched = discordUser.nick === foundUserWithDiscordId.user.username;
+            const isSuperuser = foundUserWithDiscordId.user.roles.super_user;
+            if (isDeveloper && !isMaven && !isUsernameMatched && !isBot && !isSuperuser) {
+              usersToBeEffected.push({
+                discordId: foundUserWithDiscordId.user.discordId,
+                username: foundUserWithDiscordId.user.username,
+                first_name: foundUserWithDiscordId.user.first_name,
+              });
+            }
+          }
+        } catch (error) {
```

Comment:
try/catch block. If an error occurs when fetching the user with the matching Discord ID, the code will throw a new error. However, this will cause the outer Promise.all() call to fail, even if all of the other user fetch operations were successful.

To fix this bug, you can simply remove the throw statement from the catch block. This will allow the outer Promise.all() call to succeed, even if some of the user fetch operations failed.

Reply:
Fixed it, thanks for pointing it out 🙌🏻